frog/imports/client/Wiki/index.js

```diff
@@ -182,12 +194,76 @@ class WikiComp extends React.Component<WikiCompPropsT, WikiCompStateT> {
       );
     }
 
+    // Show locked modal if the wiki is locked
+    if (
+      this.wikiDoc.data.settings?.locked &&
+      !this.wikiDoc.data.owners.find(x => x === Meteor.userId())
+    ) {
+      this.setState({ currentPageObj: null });
+      this.props.showModal(<LockedModal />);
+      return;
+    } else {
+      this.initialLoad = true;
+      this.props.hideModal();
+    }
```

Comment:
Move this code to its own function to reduce the number of things `loadWikiDoc` does

Reply:
Why would this be necessary?